### DISCUSSION TITLE
Expanding usable capacity on raidz pools by adding devices

### DISCUSSION THREAD

**Discussion by johnkwoods** [openzfs/zfs#12467] [Ideas]
We currently have a ZFS pool that has a 6-disk raidz1 vdev. We are running out of space on that zpool, and would like to expand the zpool, to add 2 more identical disks to that volume. Currently, the only known solution would be to backup the filesystems, destroy the zpool, recreate the zpool using 8-disk raidz1, then restore the filesystems.

Other technologies, such as Synology NAS, have a capability of adding a disk to a RAID-5 or RAID-6 storage pool, without having to rebuild it, or even take it offline.

It seems that this could also be done with OpenZFS, perhaps reusing/refactoring code from the sequential reconstruction feature. (I realize it's probably not that simple, or it would be done already..)

Other suggestions that would offer flexibility for raidz users:
- Ability to convert a raidz2 vdev into a raidz1 vdev, increasing the usable storage. (online rebuild)
- Ability to convert a raidz1 vdev into a raidz2 vdev, by adding a device. (online rebuild)

**Comment by IvanVolosyuk** [comment]
See https://github.com/openzfs/zfs/pull/12225

**Reply by johnkwoods** [reply]
Tremendous! Had not seen that the feature had been developed. Thank you!
